feature/gribi/otg_tests/static_lsp/static_lsp_test.go

```diff
@@ -0,0 +1,367 @@
+	iDut2Ipv4.SetAddress(ateDst.IPv4).SetGateway(dutDst.IPv4).SetPrefix(uint32(ateDst.IPv4Len))
+	iDut2Ipv6 := iDut2Eth.Ipv6Addresses().Add().SetName(ateDst.Name + ".IPv6")
+	iDut2Ipv6.SetAddress(ateDst.IPv6).SetGateway(dutDst.IPv6).SetPrefix(uint32(ateDst.IPv6Len))
+
+	// enable packet capture on this port
+	top.Captures().Add().SetName("mplsPackCapture").SetPortNames([]string{port2.Name()}).SetFormat(gosnappi.CaptureFormat.PCAP)
+
+	return top
+
+}
+
+// configureStaticLSP configures a static MPLS LSP with the provided parameters.
+func configureStaticLSP(t *testing.T, dut *ondatra.DUTDevice, lspName string, incomingLabel uint32, nextHopIP string) {
+	d := &oc.Root{}
+	mplsCfg := d.GetOrCreateNetworkInstance(deviations.DefaultNetworkInstance(dut)).GetOrCreateMpls()
+	staticMplsCfg := mplsCfg.GetOrCreateLsps().GetOrCreateStaticLsp(lspName)
+	staticMplsCfg.GetOrCreateEgress().SetIncomingLabel(oc.UnionUint32(incomingLabel))
+	staticMplsCfg.GetOrCreateEgress().SetNextHop(nextHopIP)
+	staticMplsCfg.GetOrCreateEgress().SetPushLabel(oc.Egress_PushLabel_IMPLICIT_NULL)
+
+	gnmi.Update(t, dut, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(dut)).Mpls().Config(), mplsCfg)
+
+}
+
+func createTrafficFlow(t *testing.T,
+	ate *ondatra.ATEDevice,
+	dut *ondatra.DUTDevice,
+	top gosnappi.Config,
+	label1 uint32, label2 uint32) gosnappi.Flow {
+
+	// get dut mac interface for traffic mpls flow
+	dutDstInterface := dut.Port(t, "port1").Name()
+	dstMac := gnmi.Get(t, dut, gnmi.OC().Interface(dutDstInterface).Ethernet().MacAddress().State())
+	t.Logf("DUT remote mac address is %s", dstMac)
+
+	// Create a traffic flow with MPLS
+	flowName := fmt.Sprintf("MPLS-%d-MPLS-%d", mplsLabel1, mplsLabel2)
+	mplsFlow := top.Flows().Add().SetName(flowName)
+	mplsFlow.TxRx().Port().
+		SetTxName(ate.Port(t, "port1").ID()).
+		SetRxNames([]string{ate.Port(t, "port2").ID()})
+
+	mplsFlow.Metrics().SetEnable(true)
+	mplsFlow.Rate().SetPps(500)
+	mplsFlow.Size().SetFixed(512)
+	// mplsFlow.Duration().FixedPackets().SetPackets(100000)
+	mplsFlow.Duration().Continuous()
+
+	// Set up ethernet layer.
+	eth := mplsFlow.Packet().Add().Ethernet()
+	eth.Src().SetValue(ateSrc.MAC)
+	eth.Dst().SetValue(dstMac)
+
+	// Set up MPLS layer with destination label 100.
+	mpls := mplsFlow.Packet().Add().Mpls()
+	mpls.Label().SetValue(label1)
+	mpls.BottomOfStack().SetValue(0)
+
+	// Set up MPLS layer with destination label 100.
+	mpls2 := mplsFlow.Packet().Add().Mpls()
+	mpls2.Label().SetValue(label2)
+	mpls2.BottomOfStack().SetValue(1)
+
+	ip4 := mplsFlow.Packet().Add().Ipv4()
+	ip4.Src().SetValue(ateSrc.IPv4)
+	ip4.Dst().SetValue(ateDst.IPv4)
+	ip4.Version().SetValue(4)
+
+	return mplsFlow
+
+}
+
+func ValidatePackets(t *testing.T, filename string, expextedLabel uint32) {
+	handle, err := pcap.OpenOffline(filename)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer handle.Close()
+
+	// Define tolerance
+	const tolerancePercentage = 1.0
```

Comment:
there's another `tolerance` in global const block, is this one required?

Reply:
Addressed this in latest commit